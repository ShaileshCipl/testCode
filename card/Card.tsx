import { css, merge } from "glamor";
import * as React from "react";
import Icon from "../form/icon";
import defaultStyles from "./CardStyles";

interface Props extends React.HTMLProps<HTMLDivElement> {
  children?: any;
  styles?: React.CSSProperties;
  title?: string;
  subTitle?: string;
}

const Card: React.SFC<Props> = ({ children, styles, title, ...props }: Props) => (
  <div {...css(defaultStyles, styles)} {...props} >
    {title ? <span className="cardTitleStyle">{title}</span> : null}
    <div>{children}</div>
  </div>
);

export default Card;
