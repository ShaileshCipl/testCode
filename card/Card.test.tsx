import { mount, render , shallow} from "enzyme";
import * as React from "react";
import Card from "./Card";

const title = "Approval Result";
const children = "Loan Amount";
const wrapper = mount((
  <Card title="Approval Result"/>
  ));
describe("Card component", () => {
    it("should render its title with children", () => {
      expect(wrapper.contains(<Card title="Approval Result" children={children}/>)).toMatchSnapshot();
    });

    it("should not adopt an invalid label", () => {
        const renderedComponent = shallow(<Card title="Approval Result">card content</Card>);
        expect(renderedComponent.prop("title")).toBeUndefined();
    });
  });
