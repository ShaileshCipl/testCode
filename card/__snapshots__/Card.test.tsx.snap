// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render its title with children 1`] = `false`;
